zenodo/modules/deposit/extractor.py

```diff
@@ -25,12 +25,31 @@
 """Metadata Extractor."""
 
 from __future__ import absolute_import
+from pprint import pprint
+from flask import current_app, jsonify
+from invenio_grobid.api import process_pdf_stream
+from invenio_grobid.mapping import tei_to_dict
+from invenio_grobid.errors import GrobidRequestError
+from invenio_db import db
+from invenio_files_rest.models import ObjectVersion, FileInstance
 
 
 def zenodo_metadata_extractor(version_id):
-    metadata = {
-       "title": "<Title>",
-       "description": "version_id = {}".format(version_id),
-       "version_id": version_id
-    }
+    file_id = ObjectVersion.query.filter_by(version_id=version_id).one().file_id
+    file_instance = FileInstance.get(file_id)
+    with open(file_instance.uri) as pdf_file:
```

Comment:
I use python 2.7.  I've fixed the problem following the practice here:

http://python-future.org/compatible_idioms.html#file-io-with-open